### PR TITLE
ci: bump actions version

### PR DESCRIPTION
**Pull request by tisonkun**
Before add a Dockerfile build, I'd like to make the CI workflows tidy ..

### CODE REVIEW

**Comment by daviderli614** [comment]
@tisonkun Can we fix the CI failing checks: https://github.com/GreptimeTeam/gtctl/actions/runs/8429980751/job/23085136468?pr=198

**Comment by tisonkun** [comment]
Yes. It's about empty secret that we don't set yet. I'm consulting our org admin for suggestions.

**Comment by tisonkun** [comment]
@daviderli614 fixed. PTAL

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -13,27 +10,20 @@ permissions:
...
+    name: Check typos
     runs-on: ubuntu-latest
     steps:
-      - name: Checkout Actions Repository
-        uses: actions/checkout@v3
-
-      - name: Check spelling with custom config file
-        uses: crate-ci/typos@v1.14.8
+      - uses: actions/checkout@v4
+      - uses: crate-ci/typos@v1.14.8
```
Comment (daviderli614) [inline .github/workflows/ci.yaml]: suggestion to use `crate-ci/typos@master` here, some usages in: https://github.com/crate-ci/typos/blob/master/docs/github-action.md

Comment (tisonkun) [reply .github/workflows/ci.yaml]:
```suggestion
      - uses: crate-ci/typos@master
```

**Review by daviderli614 — APPROVED** [review]
LGTM

**Pull request merged**